trieste/models/gpflow/builders.py

```diff
@@ -474,3 +497,97 @@ def build_multifidelity_autoregressive_models(
+    :cite:`perdikaris2017nonlinear` for the higher fidelities, which also have an extra input
+    dimension. Note that the initial data that the models with fidelity greater than 0 are
+    initialised with contain dummy data in this extra dimension, and so an `update` of the
+    `MultifidelityNonlinearAutoregressive` is required to propogate real data through to these
+    models.
+
+    :param dataset: The dataset to use to initialise the models
+    :param fidelities: The number of fidelities to model
+    :return: gprs: A list containing gprs that can be used for the multifidelity model
+    """
+    # Split data into fidelities
+    data = split_dataset_by_fidelity(dataset=dataset, num_fidelities=num_fidelities)
+
+    _validate_multifidelity_data_modellable(data, num_fidelities)
+
+    # Input dim requires excluding fidelity row
+    input_dim = dataset.query_points.shape[1] - 1
+
+    # Create kernels
+    kernels = _create_multifidelity_nonlinear_autoregressive_kernels(
+        gpflow.kernels.Matern32, num_fidelities, input_dim  # type:ignore
+    )
+
+    # Initialise low fidelity GP
+    gprs = [
+        GaussianProcessRegression(
+            build_gpr_with_kernel(data[0], kernel=kernels[0], likelihood_variance=1e-6)
+        )
+    ]
+
+    for fidelity in range(1, num_fidelities):
+
+        # Get query points for this fidelity
+        qps = data[fidelity].query_points
+        samples_column = tf.random.normal([qps.shape[0], 1], dtype=tf.float64)
+
+        augmented_qps = tf.concat([qps, samples_column], axis=1)
+        augmented_dataset = Dataset(augmented_qps, data[fidelity].observations)
+        gprs.append(
+            GaussianProcessRegression(
+                build_gpr_with_kernel(
+                    augmented_dataset, kernel=kernels[fidelity], likelihood_variance=1e-6
+                )
+            )
+        )
+
+    return gprs
+
+
+def _validate_multifidelity_data_modellable(data: Sequence[Dataset], num_fidelities: int) -> None:
+
+    if num_fidelities < 2:
+        raise ValueError(
+            "Invalid number of fidelities to build Multifidelity model for,"
+            f" need at least 2 fidelities, got {num_fidelities}"
+        )
+
+    for i, fidelity_data in enumerate(data):
+        if len(fidelity_data) < 2:
+            raise ValueError(
+                f"Not enough data to create model for fidelity {i},"
+                f" need at least 2 datapoints, got {len(fidelity_data)}"
+            )
+
+
+def _create_multifidelity_nonlinear_autoregressive_kernels(
+    kernel_base_class: Type[Stationary], n_fidelities: int, n_input_dims: int, ard: bool = False
+) -> Sequence[Stationary]:
+
+    dims = list(range(n_input_dims + 1))
+    if ard:
+        lengthscales = tf.ones([n_input_dims])
+    else:
+        lengthscales = 1.0
+    scale_lengthscale = 1.0
+    kernels = [kernel_base_class(lengthscales=lengthscales)]
```

Comment:
This assumes that the input space is always scaled such that a lenghtscale of 1.0 is a initial value. Might be best to have the `search_space` as input and use `_get_kernel`? would give you the priors for free.

Unless there is a good practical reason not to use those?

Reply:
We could do. We'd have to extend `_get_kernel` to work for arbitrary kernels if we also want to allow the base kernels be arbitrary for this model. There's also a little bit of trickiness of not fully knowing what the input space is for the models for fidelity 1+ because they take outputs from the lower fidelity as inputs, but maybe there's a straightforward solution or good heuristic for this

Reply:
if we leave it as it is, it will break if people use it on search_spaces that are not rescaled to approximately [0, 1], so it's probably worth doing a little bit extra work?